proxy/src/main/java/org/apache/rocketmq/proxy/service/route/TopicRouteService.java

```diff
@@ -64,10 +68,14 @@ public TopicRouteService(MQClientAPIFactory mqClientAPIFactory) {
             config.getTopicRouteServiceThreadPoolQueueCapacity()
         );
         this.mqClientAPIFactory = mqClientAPIFactory;
-        this.topicCache = CacheBuilder.newBuilder()
-            .maximumSize(config.getTopicRouteServiceCacheMaxNum())
-            .refreshAfterWrite(config.getTopicRouteServiceCacheExpiredInSeconds(), TimeUnit.SECONDS)
-            .build(new TopicRouteCacheLoader());
+
+        this.topicCache = Caffeine.newBuilder().maximumSize(config.getTopicRouteServiceCacheMaxNum()).
+            refreshAfterWrite(config.getTopicRouteServiceCacheExpiredInSeconds(), TimeUnit.SECONDS).
+            executor(cacheRefreshExecutor).build(new CacheLoader<String, MessageQueueView>() {
+                @Override public @Nullable MessageQueueView load(String topic) throws Exception {
+                    return topicRouteCacheLoader.load(topic);
```

Comment:
Does caffeine have the same error handling process as guava cache?

We must ensure that:
1. TopicRoute info should be discarded if TOPIC_NOT_EXIST occurs.
2. TopicRoute info should be kept if unexpected exceptions occur to tolerate namesrv failure.

Reply:
1. fix
2. if occur exceptions then return old value,see   https://github.com/ben-manes/caffeine/wiki/Refresh